packages/dumb/src/components/stats/total-insulin/total-insulin-stat.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2022, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+import React, { FunctionComponent, memo } from 'react'
+import styles from './total-insulin-stat.css'
+import { StatTooltip } from '../../tooltips/stat-tooltip/stat-tooltip'
+import { Box } from '@material-ui/core'
+
+interface TotalInsulinPropsData {
+  id: string
+  title: string
+  units: string
+  value: number
+  valueString: string
+}
+
+export interface TotalInsulinStatProps {
+  annotations: []
+  data: TotalInsulinPropsData[]
+  title: string
+  total: number
+}
+
+const TotalInsulinStat: FunctionComponent<TotalInsulinStatProps> = (props) => {
+  const { annotations, data, title, total } = props
+
+  const percent = (value: number): string => {
+    const res = Math.round(100 * value / total)
+    return res > 0 ? res.toString(10) : '--'
```

Comment:
Not mandatory to convert to string ? The number will be convert in the span

Reply:
The method return type is string

Reply:
and what about to type `string | number`  😅